assets/src/blocks/amp-story-text/index.js

```diff
@@ -72,12 +72,19 @@ const schema = {
 	customTextColor: {
 		type: 'string',
 	},
+	backgroundHexValue: {
+		type: 'string',
+	},
```

Comment:
Is there a better way how to store / get the hex value?

Reply:
I don't exactly see why we'd need `backgroundColor`, `customBackgroundColor`, `backgroundHexValue`, _and_ `opacity`. Can you elaborate a bit on the use here?

I thought that when we have `backgroundColor` (a color slug), we can get the hex value from that. When we have `customBackgroundColor` (a custom color picked by the user), that is already a hex value.

When we have `opacity`, we just convert either of those values to RGB and add the alpha channel.

Reply:
Do you know which method can get hex from `backgroundColor` (color slug)?

Reply:
You can use the `withColorContext` HOC from `@wordpress/components` to get the editor color palette passed as a prop. That's what `PanelColorSettings` uses too. Then you can use something like `getColorObjectByAttributeValues( props.colors, props.attributes.backgroundColor )` to get the color object (that contains the hex color value)

Reply:
Added this in the recent commits.

Reply:
Thanks, @swissspidy, works well.